src/SFA.DAS.ApprenticeAccounts/Application/Commands/ChangeEmailAddressCommand/ChangeEmailAddressCommandHandler.cs

```diff
@@ -16,7 +16,11 @@ public ChangeEmailAddressCommandHandler(IApprenticeContext apprentices)
         public async Task<Unit> Handle(ChangeEmailAddressCommand request, CancellationToken cancellationToken)
         {
             var apprentice = await _apprentices.GetById(request.ApprenticeId);
-            apprentice.UpdateEmail(new MailAddress(request.Email));
+
+            if (apprentice != null) {
+                apprentice.UpdateEmail(new MailAddress(request.Email));
+            }
+            
```

Comment:
I'm probably wrong, but I didn't think we called this handler to update the Email address. From memory I think we are using the Patch endpoint (and a patch handler). But it might call this handler inside that flow....

If we aren't using this handler (I should have removed it, apologies). 

Should we add a test to check no update occurs (which is a little unusual). And should we add logging to say no apprentice record was found.

Reply:
Thanks for your help @Pauljgraham, was scratching my head where to add this, you could well be right with the Patch endpoint (and patch handler), I'll look into that now too, thank you.